subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/resolver/IvyResourcePattern.java

```diff
@@ -51,10 +52,32 @@ public ExternalResourceName toVersionListPattern(ModuleIdentifier module, IvyArt
     }
 
     public ExternalResourceName toModulePath(ModuleIdentifier module) {
-        throw new UnsupportedOperationException("not implemented yet.");
+        ImmutableMap<String, String> attributes = ImmutableMap.of(
+            "organisation", module.getGroup(),
+            "module", module.getName(),
+            "artifact", module.getName()
+        );
+        ExternalResourceName resolve = getBase().getRoot().resolve(substituteTokens(getPathWithoutArtifactPart(), attributes));
+        return resolve;
     }
 
     public ExternalResourceName toModuleVersionPath(ModuleComponentIdentifier componentIdentifier) {
-        throw new UnsupportedOperationException("not implemented yet.");
+        ImmutableMap<String, String> attributes = ImmutableMap.of(
```

Comment:
This implementation deserves a unit test.

Reply:
done